src/lib/post.ts

```diff
@@ -0,0 +1,83 @@
+import * as fs from 'fs';
+import { join } from 'path';
+
+import matter from 'gray-matter';
+
+type Post = {
+  slug: string;
+  title: string;
+  excerpt: string;
+  content: string;
+  date: string;
+  coverImage: string;
+  ogImage: string;
+  author: string;
+};
+
+type Field =
+  | 'slug'
+  | 'title'
+  | 'excerpt'
+  | 'content'
+  | 'date'
+  | 'coverImage'
+  | 'ogImage'
+  | 'author';
+
+const BASIC_FIELDS = [
+  'title',
+  'date',
+  'slug',
+  'author',
+  'coverImage',
+  'excerpt',
+] as Field[];
+
+const postsDirectory = join(process.cwd(), '_posts');
+
+const getPostSlugs = () => {
+  return fs.readdirSync(postsDirectory);
+};
+
+const getPostBySlug = (slug: string, fields: Field[] = []): Post => {
+  const realSlug = slug.replace(/\.md$/, '');
```

Comment:
should we extract the `\.md$` part in a variable, like `SLUG_EXTENSION` or something similar?

Reply:
Extracted in https://github.com/nimblehq/til-web/pull/8/commits/b27fcb650a803e5a0588e94bb9b0f3e5267786a1